File: backend/src/main/java/fr/pedalons/dto/routes/request/RouteRequest.java
package fr.pedalons.dto.routes.request;

import fr.pedalons.common.GeoPoint;
import fr.pedalons.dto.common.asset.MediaDto;
import fr.pedalons.dto.common.request.WithVisibility;
import fr.pedalons.dto.validation.ValidateSchema;
import fr.pedalons.enums.SurfaceType;
import fr.pedalons.enums.Visibility;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import java.util.List;
import org.eclipse.microprofile.openapi.annotations.media.Schema;
import org.jspecify.annotations.Nullable;

@Schema(description = "Route update request")
@ValidateSchema
public record RouteRequest(
    @Schema(description = "Route name", required = true) @NotBlank @Size(min = 3, max = 200)
        String name,
    @Schema(description = "Media", required = true) @Valid MediaDto media,
    @Schema(description = "Surface type", required = true) SurfaceType surfaceType,
    @Schema(description = "Whether the route is publicly visible", required = true)
        Visibility visibility,
    @Nullable @Schema(description = "Points from frontend routing") List<GeoPoint> points)
    implements WithVisibility {}

File: frontend/src/api/dto/routeRequest.ts
import type { GeoPoint } from './geoPoint.ts'
import type { MediaDto } from './mediaDto.ts'
import type { SurfaceType } from './surfaceType.ts'
import type { Visibility } from './visibility.ts'

/**
 * Route update request
 */
export interface RouteRequest {
  /**
   * Route name
   * @minLength 3
   * @maxLength 200
   * @pattern \S
   */
  name: string
  /** Media */
  media: MediaDto
  /** Surface type */
  surfaceType: SurfaceType
  /** Whether the route is publicly visible */
  visibility: Visibility
  /** Points from frontend routing */
  points?: GeoPoint[]
}
